eva-accession-import-automation/create_fasta_from_assembly_report.sh

```diff
@@ -22,9 +22,24 @@ for genbank_contig in `grep -v -e "^#" ${assembly_report} | cut -f5`;
 do
     echo ${genbank_contig}
 
-    # Download each GenBank accession in the assembly report from ENA into a separate file
-    # Delete the accession prefix from the header line
-    wget -q -O - "https://www.ebi.ac.uk/ena/browser/api/fasta/${genbank_contig}" | sed 's/ENA|.*|//g' > ${output_folder}/${genbank_contig}
+    # make $? return the error code of any failed command in the pipe, instead of the last one only
+    set -o pipefail
+    times_wget_failed=0
+    max_allowed_attempts=5
+    while [ $times_wget_failed -lt $max_allowed_attempts ]
+    do
+        # Download each GenBank accession in the assembly report from ENA into a separate file
+        # Delete the accession prefix from the header line
+        wget -q -O - "https://www.ebi.ac.uk/ena/browser/api/fasta/${genbank_contig}" | sed 's/ENA|.*|//g' > ${output_folder}/${genbank_contig}
+        whole_pipe_result=$?
+        if [ $whole_pipe_result -eq 0 ]
+        then
+            # it was correctly downloaded
+            break
+        fi
+        echo Download for ${genbank_contig} failed. Retrying...
```

Comment:
What do you think about only printing the retrying message the first time it has to retry? Otherwise we will get a lot of repetitive messages. We will have this retrying messages a lot for some species specially in the cases when the contigs belong to a WGS as those are not available through the API.

So I would suggest having this
```
CM000815.5
CM000816.5
Download for CM000816.5 failed. Retrying...
FASTA sequence not available for CM000816.5
CM000817.5
Download for CM000817.5 failed. Retrying...
FASTA sequence not available for CM000817.5
```
instead of this
```
CM000815.5
CM000816.5
Download for CM000816.5 failed. Retrying...
Download for CM000816.5 failed. Retrying...
Download for CM000816.5 failed. Retrying...
Download for CM000816.5 failed. Retrying...
Download for CM000816.5 failed. Retrying...
FASTA sequence not available for CM000816.5
CM000817.5
Download for CM000817.5 failed. Retrying...
Download for CM000817.5 failed. Retrying...
Download for CM000817.5 failed. Retrying...
Download for CM000817.5 failed. Retrying...
Download for CM000817.5 failed. Retrying...
FASTA sequence not available for CM000817.5
```

At least while we find out if there is a way to know the contig belong to a WGS in which case we wouldn't need to try downloading it through the API but going straight to the FTP

Reply:
makes sense. if the contig was successfully downloaded at the second of third time, it won't print "sequence not available"